client/data/domains/use-get-domains-query.ts

```diff
@@ -1,11 +1,14 @@
 import { useQuery } from 'react-query';
 import wpcom from 'calypso/lib/wp';
-import type { ResponseDomain } from 'calypso/lib/domains/types';
 import type { UseQueryOptions } from 'react-query';
 
 export const getCacheKey = ( siteId: number | null ) => [ 'sites', siteId, 'domains' ];
 
-type UseGetDomainsQueryData = ResponseDomain[];
+export type Domain = {
+	domain: string;
+	registration_date: string;
+};
+type UseGetDomainsQueryData = Domain[];
```

Comment:
the [UseGetDomainsQueryData](https://github.com/Automattic/wp-calypso/blob/trunk/client/data/domains/use-get-domains-query.ts#L8) was previously using the [ResponseDomain type](https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/domains/types.ts#L48). However, this was incorrect because the `ResponseDomain` type has all of the properties camel cased but the actual response object from the API has these properties snake cased so the properties weren't defined correctly. The reason this wasn't breaking is that the [useGetDomainsQuery](https://github.com/Automattic/wp-calypso/blob/trunk/client/data/domains/use-get-domains-query.ts#L10) has only been accessing these properties [in JavaScript files](https://github.com/Automattic/wp-calypso/blob/trunk/client/state/sites/domains/assembler.js#L41) and not TypeScript. That's why I created this new `Domain` type to define the properties in camel case. I only defined the `domain` and `registration_date` properties because these are the only ones I am using right now and we don't need to define the rest of them.

Reply:
I ran into the same issue yesterday with the `ResponseDomain` type. I found a solution for it using an existing function that will create the correct ResponseDomain object for us. That will solve the camelCase issue.